app/Api/Middleware/LogApiRequest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Api\Middleware;
+
+use App\Models\ApiLogEntry;
+use Closure;
+use Illuminate\Http\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class LogApiRequest
+{
+    public function __construct(
+        private readonly string $apiVersion = 'v2'
+    ) {
+    }
+
+    public function handle(Request $request, Closure $next, ?string $apiVersion = null): Response
+    {
+        $startTime = microtime(true);
+
+        $version = $apiVersion ?? $this->apiVersion;
+
+        $response = $next($request);
+
+        $user = $request->user();
+
+        ApiLogEntry::logRequest(
+            $version,
+            $user?->id,
+            $request->path(),
+            $request->method(),
+            $response->getStatusCode(),
+            $this->calculateResponseTime($startTime),
+            $request->ip(),
+            $request->userAgent(),
+            $this->sanitizeRequestData($request),
+            $this->getErrorMessage($response)
+        );
+
+        return $response;
```

Comment:
Consider injecting the `Content-Length` header in the response. See https://github.com/RetroAchievements/RAWeb/pull/3634#discussion_r2148064509

Reply:
Wrote a small middleware to handle this for us automatically. Success responses now contain a `Content-Length`:
<img width="671" height="301" alt="Screenshot 2025-07-31 at 8 15 02 PM" src="https://github.com/user-attachments/assets/6085c117-421e-4640-a867-711c4955d33e" />

The value mirrors what's stored in the new `response_size_bytes` column in `api_logs`.